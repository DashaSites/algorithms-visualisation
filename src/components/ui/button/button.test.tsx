import renderer from "react-test-renderer";
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from "./button";


describe("Button component testing", () => {

  it("Is button with text displayed correctly", () => {
    const tree = renderer.create(<Button text="Какой-то текст" />).toJSON();
    expect(tree).toMatchSnapshot();
  });


  it("Is button without text displayed correctly", () => {
    const tree = renderer.create(<Button text="" />).toJSON();
    expect(tree).toMatchSnapshot();
  });


  it("Is disabled button displayed correctly", () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });


  it("Is button with loader displayed correctly", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });


  it("Is a callback function being called correctly with a buttonclick", () => {
    const someFunction = jest.fn(); // моковая функция
    render(<Button onClick={someFunction} text="Сделать" />);
    const button = screen.getByText("Сделать");
    fireEvent.click(button);
    // if there are some precise arguments given to `someFunction` maybe 
    // use `toHaveBeenCalledWith` instead
    expect(someFunction).toHaveBeenCalled();
  });
});

