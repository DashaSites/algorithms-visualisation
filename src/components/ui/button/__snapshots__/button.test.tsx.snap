// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component testing Is button with loader displayed correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src={{}}
  />
</button>
`;

exports[`Button component testing Is button with text displayed correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Какой-то текст
  </p>
</button>
`;

exports[`Button component testing Is button without text displayed correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component testing Is disabled button displayed correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
